compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -132,14 +132,38 @@ impl<'a> Parser<'a> {
     /// The difference from `parse_ty` is that this version allows `...`
     /// (`CVarArgs`) at the top level of the type.
     pub(super) fn parse_ty_for_param(&mut self) -> PResult<'a, P<Ty>> {
-        self.parse_ty_common(
+        let ty = self.parse_ty_common(
             AllowPlus::Yes,
             AllowCVariadic::Yes,
             RecoverQPath::Yes,
             RecoverReturnSign::Yes,
             None,
             RecoverQuestionMark::Yes,
-        )
+        )?;
+
+        // Recover a trailing `= EXPR` if present.
+        if self.may_recover() {
+            let snapshot = self.create_snapshot_for_diagnostic();
```

Comment:
This clones the entire parser in the happy path which may incur a perf hit. Any reason why you don't gate the snapshot behind `check_noexpect(Eq)`/`==Eq` (followed by `bump`)?

Reply:
Oh, d'oh. Of course lmao. Yeah. I was thinking about doing this then just forgor.

Reply:
nw, r=me then